File: Day-03/C++/main.cpp
#include <bits/stdc++.h>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false); 
    cin.tie(NULL);

    vector<vector<int>> v;
    int a, b, c, possible = 0, possible2 = 0;
    while (cin >> a >> b >> c) {
        v.push_back({a, b, c});
        if (a + b + c > 2 * max(max(a, b), c)) {
            possible++;
        }
    }

    for (int i = 0; i < (int) v.size(); i += 3) {
        for (int j = 0; j < 3; j++) {
            if (v[i][j] + v[i + 1][j] + v[i + 2][j] > 2 * max(max(v[i][j], v[i + 1][j]), v[i + 2][j])) {
                possible2++;
            }
        }
    }

    cout << "Part 1: " << possible << endl;
    cout << "Part 2: " << possible2 << endl;

    return 0;
}

File: Day-03/Python/main.py
import sys

possible = 0
m = []
for line in sys.stdin:
    a, b, c = list(map(int, line.strip().split()))
    possible += int(a + b + c > 2 * max(a, b, c))
    m.append([a, b, c])
print("Part 1:", possible)

possible2 = 0
for i in range(0, len(m), 3):
    for j in range(3):
        possible2 += int(m[i][j] + m[i + 1][j] + m[i + 2][j] > 2 * max(m[i][j], m[i + 1][j], m[i + 2][j]))
print("Part 2:", possible2)